src/cas/flatten_data_to_anndata.py

```diff
@@ -43,67 +47,107 @@ def is_list_of_strings(var):
     return isinstance(var, list) and all(isinstance(item, str) for item in var)
 
 
-def flatten(json_file_path, anndata_file_path, validate, output_file_path):
+def flatten(json_file_path, anndata_file_path, output_file_path):
     """
      Processes and integrates information from a JSON file and an AnnData (Annotated Data) file, creating a new AnnData
      object that incorporates the metadata. The resulting AnnData object is then saved to a new file.
 
     Args:
         json_file_path: The path to the CAS json file.
         anndata_file_path: The path to the AnnData file.
-        validate: Boolean to determine if validation checks will be performed before writing to the output AnnData file.
         output_file_path: Output AnnData file name.
     """
     input_json = read_json_file(json_file_path)
-    input_anndata = read_anndata_file(anndata_file_path)
 
-    flatten_cas_object(input_json, input_anndata, validate, output_file_path)
+    flatten_cas_object(input_json, anndata_file_path, output_file_path)
 
 
-def flatten_cas_object(input_json, input_anndata, validate, output_file_path):
+# @profile
+def flatten_cas_object(input_json, anndata_file_path, output_file_path):
     """
      Processes and integrates information from a JSON file and an AnnData (Annotated Data) file, creating a new AnnData
      object that incorporates the metadata. The resulting AnnData object is then saved to a new file.
 
     Args:
         input_json: CAS json file.
-        input_anndata: The AnnData object.
-        validate: Boolean to determine if validation checks will be performed before writing to the output AnnData file.
+        anndata_file_path: The path to the AnnData file.
         output_file_path: Output AnnData file name.
     """
-    # obs
-    annotations = input_json[ANNOTATIONS]
+    if output_file_path:
+        shutil.copy(anndata_file_path, output_file_path)
+        anndata_file_path = output_file_path
 
+    annotations = input_json[ANNOTATIONS]
     parent_cell_ids = collect_parent_cell_ids(input_json)
 
+    with h5py.File(anndata_file_path, "r+") as f:
```

Comment:
Existing functions in the `file_utils.py` are useful but can we add this operations to the file_utils as well? This will be a higher level function and easier to reuse
```
def update_hdf5(anndata_file_path, obs_data=None, uns_data=None):
```

We can do `process_annotations` and `generate_uns_json` operations before this code block.